libcloud/container/drivers/lxd.py

```diff
@@ -0,0 +1,1269 @@
+
+        if key_file:
+
+            if not cert_file:
+                # LXD tls authentication-
+                # We pass two files, a key_file with the
+                # private key and cert_file with the certificate
+                # libcloud will handle them through LibcloudHTTPSConnection
+
+                raise LXDAPIException(message='Need both private key and'
+                                              ' certificate files for '
+                                              'tls authentication')
+
+            self.connectionCls = LXDtlsConnection
+            self.key_file = key_file
+            self.cert_file = cert_file
+            self.certificate_validator = certificate_validator
+            secure = True
+
+        if host.startswith('https://'):
+            secure = True
+
+        host = strip_http_prefix(host=host)
+
+        super(LXDContainerDriver, self).__init__(key=key, secret=secret,
+                                                 secure=secure, host=host,
+                                                 port=port,
+                                                 key_file=key_file,
+                                                 cert_file=cert_file)
+
+        if ca_cert:
+            self.connection.connection.ca_cert = ca_cert
+        else:
+            # do not verify SSL certificate
+            self.connection.connection.ca_cert = False
+
+        self.connection.secure = secure
+        self.connection.host = host
+        self.connection.port = port
+        self.version = self._get_api_version()
+
+    def ex_get_api_endpoints(self):
+        """
+        Description: List of supported APIs
+        Authentication: guest
+        Operation: sync
+        Return: list of supported API endpoint URLs
+
+        """
+        response = self.connection.request("/")
+        response_dict = response.parse_body()
+        assert_response(response_dict=response_dict, status_code=200)
+        return response_dict["metadata"]
+
+    def ex_get_server_configuration(self):
+        """
+
+        Description: Server configuration and environment information
+        Authentication: guest, untrusted or trusted
+        Operation: sync
+        Return: Dict representing server state
+
+        The returned configuration depends on whether the connection
+        is trusted or not
+        :rtype: :class: .LXDServerInfo
+
+        """
+        response = self.connection.request("/%s" % (self.version))
+        response_dict = response.parse_body()
+        assert_response(response_dict=response_dict, status_code=200)
+        meta = response_dict["metadata"]
+        return LXDServerInfo.build_from_response(metadata=meta)
+
+    def deploy_container(self, name, image, cluster=None,
+                         parameters=None, start=True,
+                         architecture=default_architecture,
+                         profiles=[default_profiles],
```

Comment:
You don't appear to be mutating ``profiles`` inside the function body, but it would still be safer and less error prone to avoid mutable defaults (https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments) and something like ``profiles = None``
and inside the body ``profiles = proxies or [default_profiles]``).